File: 프로그래머스/2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    arr = []
    
    for i in range(len(skill_trees)):
        check1 = 0
        check2 = 0
        for j in range(len(skill_trees[i])):
            if skill_trees[i][j] in skill:
                check2 += 1
            if skill_trees[i][j] == skill[check1]:
                check1 += 1
                if check1 == len(skill):
                    break # 스킬을 모두 마스터한 상태를 의미한다
        
        if check1 == check2:
            answer += 1
            
    return answer

File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
import java.util.*;

class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;

        for (String tree : skill_trees) {
            StringBuilder filtered = new StringBuilder();

            for (int i = 0; i < tree.length(); i++) {
                char c = tree.charAt(i);
                if (skill.indexOf(c) != -1) {
                    filtered.append(c);
                }
            }

            if (skill.startsWith(filtered.toString())) {
                answer++;
            }
        }

        return answer;
    }
}